vnext/Mso/future/futureWinRT.h

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+#ifndef MSO_FUTURE_FUTUREWINRT_H
+#define MSO_FUTURE_FUTUREWINRT_H
+
+#include <errorCode/hresultErrorProvider.h>
+#include <future/future.h>
+#include <winrt/Windows.Foundation.h>
+
+namespace Mso {
+
+struct AsyncActionFutureAdapter : winrt::implements<
```

Comment:
>AsyncActionFutureAdapter [](start = 7, length = 24)

Just checking if we have a convention like is ocmmon in C# to have one type per file and match the filename to the type. 
Here I see fileName: `futureWinRT.h`, with type struct `AsyncActionFutureAdapter`

Reply:
I chose the file name for the 'growth'. In C++ we often have multiple types in the same file. 
The WinRT has four async interfaces based on the IAsyncInfo: IAsyncAction, IAsyncOperation, IAsyncActionWithProgress, an IAsyncOperationWithProgress. This file must contain conversion to all four of them and then conversion back from them. I just did a required bare minimum at this point.

---
In reply to: [432968554](https://github.com/microsoft/react-native-windows/pull/5071#discussion_r432968554) [](ancestors = 432968554)